src/room/InCallView.tsx

```diff
@@ -259,14 +288,109 @@ export const InCallView: FC<InCallViewProps> = subscribe(
       [fullscreenItem, noControls, items],
     );
 
-    const Grid =
-      items.length > 12 && layout === "grid" ? NewVideoGrid : VideoGrid;
-
     const prefersReducedMotion = usePrefersReducedMotion();
 
-    // This state is lifted out of NewVideoGrid so that layout states can be
-    // restored after a layout switch or upon exiting fullscreen
-    const layoutStates = useLayoutStates();
+    const [settingsModalOpen, setSettingsModalOpen] = useState(false);
+    const [settingsTab, setSettingsTab] = useState(defaultSettingsTab);
+
+    const openSettings = useCallback(
+      () => setSettingsModalOpen(true),
+      [setSettingsModalOpen],
+    );
+    const closeSettings = useCallback(
+      () => setSettingsModalOpen(false),
+      [setSettingsModalOpen],
+    );
+
+    const openProfile = useCallback(() => {
+      setSettingsTab("profile");
+      setSettingsModalOpen(true);
+    }, [setSettingsTab, setSettingsModalOpen]);
+
+    const [headerRef, headerBounds] = useMeasure();
+    const [footerRef, footerBounds] = useMeasure();
+    const gridBounds = useMemo(
+      () => ({
+        width: footerBounds.width,
+        height: bounds.height - headerBounds.height - footerBounds.height,
+      }),
+      [
+        footerBounds.width,
+        bounds.height,
+        headerBounds.height,
+        footerBounds.height,
+      ],
+    );
+    const gridBoundsObservable = useObservable(gridBounds);
+    const floatingAlignment = useInitial(
+      () => new BehaviorSubject(defaultAlignment),
+    );
+    const { fixed, scrolling } = useInitial(() =>
+      gridLayoutSystems(state(gridBoundsObservable), floatingAlignment),
+    );
+
+    const setGridMode = useCallback(
+      (mode: GridMode) => {
+        setLegacyLayout(mode);
+        vm.setGridMode(mode);
+      },
+      [setLegacyLayout, vm],
+    );
+
+    const showSpeakingIndicators =
+      layout.type === "spotlight" ||
+      (layout.type === "grid" && layout.grid.length > 2);
+
+    const SpotlightTileView = useMemo(
+      () =>
+        forwardRef<HTMLDivElement, TileProps<MediaViewModel[], HTMLDivElement>>(
+          function SpotlightTileView(
```

Comment:
Nitpick, I think this can be an anonymous function
```suggestion
          function (
```

Reply:
It can't unfortunately, because then the component will be missing a display name for React devtools (which the linter rightfully complains about)